### PR TITLE
update requirement_guide

### PR DESCRIPTION
**Pull request by ramsayleung**
as title

### CODE REVIEW

File: document_guides/requirement_guide.md [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
 
 the requirement should contains the fellow content:
 
-### The purpose of this project
-we should describe the purpose of this project, why we start this project.
- 
-### The output of this project
-this may be a jar file, jar file, docker image etc. 
+## 1. Introduction 
+a short description about project
```
Comment (kaka89) [inline document_guides/requirement_guide.md]: A

File: document_guides/requirement_guide.md [hunk 2]
```diff
@@ -1,19 +1,42 @@
...
+Thus, you should explicitly point out the supported features of your project as
+example show
+
+## 3. Requirement
+it is mainly about technical requirement which pertains to the technical aspects
+that project should fulfill. This requirement typically refer to how project is
+built, such as what kind of standard it should meet, Taking `category` project
+for
+example,
+[category requirement](https://github.com/reactivesw/category/tree/master/docs)
```
Comment (kaka89) [inline document_guides/requirement_guide.md]: which doc you want to refer to?

File: document_guides/requirement_guide.md [hunk 3]
```diff
@@ -1,19 +1,42 @@
...
+demonstrates the basic requirement about category, such as parameters whether it
+is required or not, relation between different category. Then it shows the
+different requirement for features it provides. 
+
 
-### The king of service provided
-Rest API, grpc api, or web content etc.
 
-### The detail of the service or content that this project provided
```
Comment (kaka89) [inline document_guides/requirement_guide.md]: My opinion:

# Title --- usually the service name
*description of this micro-service* --- the purpose of this service
# The Services or futures provided
*services provided*
# Key points
*key points of the service*, for example: `password must be at least 8 characters`

# Other services needed, and why.

File: document_guides/requirement_guide.md [hunk 4]
```diff
@@ -1,19 +1,80 @@
...
-### The output of this project
-this may be a jar file, jar file, docker image etc. 
+   if this micro service is single usage, one document is enough.
 
-### The king of service provided
-Rest API, grpc api, or web content etc.
+* `admin_requirement.md`
 
-### The detail of the service or content that this project provided
+  requirement document for admin-web.
```
Comment (kaka89) [inline document_guides/requirement_guide.md]: for admin， not admin-web

File: document_guides/requirement_guide.md [hunk 5]
```diff
@@ -1,19 +1,80 @@
...
 
+* `customer_requirement.md`
 
+  requrement document for customer-web.
```
Comment (kaka89) [inline document_guides/requirement_guide.md]: for customer, not customer-web

File: document_guides/requirement_guide.md [hunk 6]
```diff
@@ -1,19 +1,80 @@
...
 
+## 1. Document Structure
+
+All requirement document is `markdown` file, and the structure should be similar, here is schema:
```
Comment (kaka89) [inline document_guides/requirement_guide.md]: All requirement documents should be  `markdown` file, and the structure should be similar, here is schema:

File: document_guides/requirement_guide.md [hunk 7]
```diff
@@ -1,19 +1,80 @@
 # Requirement Guide
```
Comment (kaka89) [inline document_guides/requirement_guide.md:1]: capital for first character

File: document_guides/requirement_guide.md (hunk 6 above)
Comment (davis999) [reply document_guides/requirement_guide.md]: done.

File: document_guides/requirement_guide.md (hunk 5 above)
Comment (davis999) [reply document_guides/requirement_guide.md]: done.

File: document_guides/requirement_guide.md (hunk 4 above)
Comment (davis999) [reply document_guides/requirement_guide.md]: done.

File: document_guides/requirement_guide.md (hunk 7 above)
Comment (davis999) [reply document_guides/requirement_guide.md:1]: done.

**Review by kaka89 — APPROVED** [review]

**Review by kaka89 — APPROVED** [review]

**Pull request merged**